components/tikv_util/src/mpsc/batch.rs

```diff
@@ -352,8 +352,26 @@ mod tests {
 
     #[test]
     fn test_batch_receiver() {
+        struct TestBatchReceiver<T, E, I: Fn() -> E, C: Fn(&mut E, T)> {
```

Comment:
Can you explain why it's unstable and why the fix is necessary?

Reply:
If the spawned thread is slow, the receiver could have not been polled before `send` is called in the another thread. So even if the sender doesn't notify the receiver, it still could be polled in the spawned thread, which will make the case fail.

Reply:
How about using `futures::select` instead of custom struct? And using channel seems more appropriate than atomic counter, as it has to wait.